File: Example/FFPopup_Example/Classes/Home/FFHomeViewController.m
//
//  FFHomeViewController.m
//  FFPopup_Example
//
//  Created by JonyFang on 2018/11/29.
//  Copyright © 2018年 JonyFang. All rights reserved.
//

#import "FFHomeViewController.h"
#import "FFPopup.h"

#define K_SCREEN_WIDTH [[UIScreen mainScreen] bounds].size.width
#define K_SCREEN_HEIGHT [[UIScreen mainScreen] bounds].size.height

@interface FFHomeViewController ()
@property (nonatomic, strong) UIButton *popupButton;
@end

@implementation FFHomeViewController

#pragma mark - Life Cycle
- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupViews];
}

#pragma mark - Private Methods
- (void)setupViews {
    self.view.backgroundColor = [UIColor whiteColor];
    [self.view addSubview:self.popupButton];
    /// Make Constraints
    [_popupButton mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.equalTo(self.view.mas_bottom).offset(-22);
        make.right.equalTo(self.view.mas_right).offset(-44);
        make.left.equalTo(self.view.mas_left).offset(44);
        make.height.mas_equalTo(44);
    }];
}

- (void)popupButtonClicked:(UIButton *)button {
    CGFloat width = K_SCREEN_WIDTH - 40.0;
    CGFloat height = width * 3.0 / 4.0;
    UIView *contentView = [[UIView alloc] init];
    contentView.backgroundColor = [UIColor blueColor];
    contentView.frame = CGRectMake(0.0, 0.0, width, height);
    contentView.layer.cornerRadius = 6.0;
    contentView.layer.masksToBounds = YES;
    FFPopup *popup = [FFPopup popupWithContentView:contentView showType:FFPopupShowType_BounceInFromTop dismissType:FFPopupDismissType_BounceOutToBottom maskType:FFPopupMaskType_Dimmed dismissOnBackgroundTouch:YES dismissOnContentTouch:NO];
    [popup show];
}

#pragma mark - Properties
- (UIButton *)popupButton {
    if (!_popupButton) {
        _popupButton = [UIButton new];
        _popupButton.backgroundColor = UIColor.blueColor;
        _popupButton.titleLabel.font = [UIFont boldSystemFontOfSize:16.0];
        [_popupButton setTitle:@"Show Popup" forState:UIControlStateNormal];
        [_popupButton addTarget:self action:@selector(popupButtonClicked:) forControlEvents:UIControlEventTouchUpInside];
        _popupButton.layer.cornerRadius = 22;
        _popupButton.layer.masksToBounds = YES;
    }
    return _popupButton;
}

@end

File: Demo/FFPopup_Swift/Classes/Controllers/FFHomeViewController.swift
//
//  FFHomeViewController.swift
//  FFPopup_Swift
//
//  Created by JonyFang on 2019/4/28.
//  Copyright © 2019 JonyFang. All rights reserved.
//

import UIKit
import SnapKit

class FFHomeViewController: UIViewController {

    // MARK: Life Cycle
    override func viewDidLoad() {
        super.viewDidLoad()
        setupViews()
    }
    
    // MARK: Private Methods
    fileprivate func setupViews() {
        view.backgroundColor = .white
        view.addSubview(popButton)
        
        popButton.snp.makeConstraints {
            $0.centerY.equalTo(view.snp_centerY);
            $0.centerX.equalTo(view.snp_centerX);
            $0.width.height.equalTo(200);
        }
    }
    
    @objc fileprivate func showPopup() {
        let popup = FFPopup(contentView: self.alertView, showType: .bounceIn, dismissType: .shrinkOut, maskType: .dimmed, dismissOnBackgroundTouch: true, dismissOnContentTouch: false)
        let layout = FFPopupLayout(horizontal: .center, vertical: .center)
        popup.show(layout: layout)
    }
    
    // MARK: Properties
    fileprivate lazy var alertView: BLCustomContentView = {
        let w = UIScreen.main.bounds.size.width - 30
        let h = w * 3 / 4
        let frame = CGRect(x: 0, y: 0, width: w, height: h)
        let view = BLCustomContentView(frame: frame)
        view.backgroundColor = .purple
        return view
    }()
    fileprivate lazy var popButton: UIButton = {
        let button = UIButton()
        button.setTitle("Pop", for: .normal)
        button.setTitle("Pop", for: .selected)
        button.setTitleColor(UIColor.white, for: .normal)
        button.setTitleColor(UIColor.white, for: .selected)
        button.backgroundColor = UIColor.blue
        button.layer.cornerRadius = 100
        button.layer.masksToBounds = true
        button.addTarget(self, action: #selector(showPopup), for: .touchUpInside)
        return button
    }()

}

